plasma_framework/contracts/src/exits/payment/controllers/PaymentChallengeIFEOutputSpent.sol

```diff
@@ -0,0 +1,158 @@
+
+        //pay bond to challenger
+        msg.sender.transfer(PIGGYBACK_BOND);
+        emit InFlightExitOutputBlocked(msg.sender, keccak256(args.inFlightTx), outputIndex);
+    }
+
+    function verifyInFlightTransactionStandardFinalized(
+        Controller memory controller,
+        PaymentInFlightExitRouterArgs.ChallengeOutputSpent memory args
+    )
+        private
+        view
+    {
+        UtxoPosLib.UtxoPos memory utxoPos = UtxoPosLib.UtxoPos(args.outputUtxoPos);
+        TxFinalization.Verifier memory finalizationVerifier = TxFinalization.moreVpVerifier(
+            controller.framework,
+            args.inFlightTx,
+            utxoPos.txPos(),
+            args.inFlightTxInclusionProof
+        );
+
+        require(TxFinalization.isStandardFinalized(finalizationVerifier), "In-flight transaction not finalized");
+    }
+
+    function verifyOutputType(
+        Controller memory controller,
+        PaymentInFlightExitRouterArgs.ChallengeOutputSpent memory args
+    )
+        private
+        view
+    {
+        UtxoPosLib.UtxoPos memory utxoPos = UtxoPosLib.UtxoPos(args.outputUtxoPos);
+        uint16 outputIndex = UtxoPosLib.outputIndex(utxoPos);
+        WireTransaction.Output memory output = WireTransaction.getOutput(args.inFlightTx, outputIndex);
+        OutputGuardModel.Data memory outputGuardData = OutputGuardModel.Data({
+            guard: output.outputGuard,
+            outputType: args.outputType,
+            preimage: args.outputGuardPreimage
+        });
+        IOutputGuardHandler handler = controller.outputGuardHandlerRegistry
+                                                .outputGuardHandlers(args.outputType);
+
+        require(address(handler) != address(0),
+            "Does not have outputGuardHandler registered for the output type");
+
+        require(handler.isValid(outputGuardData),
+            "Some of the output guard related information is not valid");
+    }
+
+    function verifyChallengingTransactionSpendsOutput(
+        Controller memory controller,
+        PaymentInFlightExitRouterArgs.ChallengeOutputSpent memory args,
+        PaymentExitDataModel.InFlightExit storage ife
+    )
+        private
+        view
+    {
+        UtxoPosLib.UtxoPos memory utxoPos = UtxoPosLib.UtxoPos(args.outputUtxoPos);
+        uint16 outputIndex = UtxoPosLib.outputIndex(utxoPos);
+        PaymentTransactionModel.Transaction memory inFlightTx = PaymentTransactionModel.decode(args.inFlightTx);
+        uint256 challengingTxType = WireTransaction.getTransactionType(args.challengingTx);
+        bytes32 outputId = ife.outputs[outputIndex].outputId;
+
+        ISpendingCondition condition = controller.spendingConditionRegistry.spendingConditions(
+            args.outputType,
+            challengingTxType
+        );
+        require(address(condition) != address(0), "Spending condition contract not found");
+
+        bool isSpentBySpendingTx = condition.verify(
```

Comment:
The interface input seems not valid to the interface though?

Reply:
I think you must have cheked the code before I uploaded merge commit.

Reply:
ah, no it wasn't like this when I check the code XDD